EventListener/CommentVoteScoreListener.php

```diff
@@ -23,18 +25,52 @@
  */
 class CommentVoteScoreListener implements EventSubscriberInterface
 {
+    private $voteManager;
+    private $voteUnique;
+
+    /**
+     * @param bool $voteUnique
+     */
+    public function __construct(VoteManagerInterface $voteManager, $voteUnique = false)
+    {
+        $this->voteManager = $voteManager;
+        $this->voteUnique = $voteUnique;
+    }
+
     public function onVotePersist(VoteEvent $event)
     {
         $vote = $event->getVote();
         $comment = $vote->getComment();
-        $comment->incrementScore($vote->getValue());
+        if (!$comment instanceof VotableCommentInterface) {
+            return;
+        }
+
+        if (true === $this->voteManager->isNewVote($vote)) {
+            $comment->incrementScore($vote->getValue());
+        } else if (false === $this->voteManager->isNewVote($vote) && true === $this->voteUnique) {
+            $comment->incrementScore($vote->getValue() * 2);
```

Comment:
When updating vote, we must double the value for a correct update total score.
Default `0` score, vote to positive now `1` update to negative should become `-1`

Do you have another idea how to do it better?